src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,522 @@ const ActorInit En_Bat_InitVars = {
+        if (this->actionFunc == EnBat_DiveAttack) {
+            Actor_UpdateBgCheckInfo(play, &this->actor, 12.0f, 15.0f, 50.0f, 5);
+        } else if ((this->actionFunc != EnBat_FlyIdle) ||
+                   ((this->actor.xzDistToPlayer < 400.0f) && (this->actor.projectedPos.z > 0.0f))) {
+            if (this->paramFlags & BAD_BAT_PARAMFLAG_0) {
+                Actor_UpdateBgCheckInfo(play, &this->actor, 12.0f, 15.0f, 50.0f, 5);
+            } else {
+                Actor_UpdateBgCheckInfo(play, &this->actor, 12.0f, 15.0f, 50.0f, 4);
+            }
+        } else {
+            Math_Vec3f_Copy(&this->actor.prevPos, &prevPos);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A438F8.s")
+        Actor_SetFocus(&this->actor, this->actor.shape.yOffset * 0.01f);
+        this->collider.dim.worldSphere.center.x = this->actor.focus.pos.x;
+        this->collider.dim.worldSphere.center.y = this->actor.focus.pos.y;
+        this->collider.dim.worldSphere.center.z = this->actor.focus.pos.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A4392C.s")
+    if (this->collider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->actionFunc == EnBat_DiveAttack) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A43CA0.s")
+    if (this->drawDmgEffAlpha > 0.0f) {
+        if (this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+            Math_StepToF(&this->drawDmgEffAlpha, 0.0f, 0.05f);
+            this->drawDmgEffScale = (this->drawDmgEffAlpha + 1.0f) * 0.225f;
+            this->drawDmgEffScale = CLAMP_MAX(this->drawDmgEffScale, 0.45f);
+        } else if (!Math_StepToF(&this->drawDmgEffFrozenSteamScale, 0.45f, 0.45f / 40.0f)) {
+            func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A43CE8.s")
+void EnBat_Draw(Actor* thisx, PlayState* play) {
+    EnBat* this = THIS;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A43F60.s")
+    // Draw body and wings
+    if (this->actor.projectedPos.z > 0.0f) {
+        OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A44114.s")
+        gfx = POLY_OPA_DISP;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bat/func_80A44294.s")
+        gSPDisplayList(&gfx[0], &sSetupDL[6 * 25]);
+        gSPMatrix(&gfx[1], Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(&gfx[2], gBadBatSetupDL);
+        gSPDisplayList(&gfx[3], gBadBatBodyDL);
+        gSPDisplayList(&gfx[4], sWingsDLs[this->animationFrame]);
```

Comment:
... there are different DL per animation frame? that was preferable to limb angles somehow?

Reply:
It's *way* cheaper. Look at the Keese skeleton and imagine having 20 of those flying around at once.